cub/test/catch2_test_device_reduce.cu

```diff
@@ -0,0 +1,464 @@
+    result_t gpu_result = out_result[0];
+    REQUIRE(expected_result[0] == gpu_result.value);
+    REQUIRE((expected_result - host_items.cbegin()) == gpu_result.key);
+  }
+}
+
+// TEST_TYPES == 4
+#elif TEST_TYPES == 4
+
+CUB_TEST("Device sum works for big indices", "[reduce][device]")
+{
+  test_big_indices_helper<std::size_t, std::uint32_t>(1ull << 30);
+  test_big_indices_helper<std::size_t, std::uint32_t>(1ull << 31);
+  test_big_indices_helper<std::size_t, std::uint32_t>((1ull << 32) - 1);
+  test_big_indices_helper<std::size_t, std::uint64_t>(1ull << 33);
+}
+
+CUB_TEST("Device reduce arg{min,max} works with inf items", "[reduce][device]")
+{
+  using in_t     = float;
+  using offset_t = int;
+  using out_t    = cub::KeyValuePair<offset_t, in_t>;
+
+  const int n     = 10;
+  const float inf = ::cuda::std::numeric_limits<float>::infinity();
+
+  thrust::device_vector<out_t> out(1);
+  out_t *d_out = thrust::raw_pointer_cast(out.data());
+
+  /**
+   * ArgMin should return max value for empty input. This interferes with
+   * input data containing infinity values. This test checks that ArgMin
+   * works correctly with infinity values.
+   */
+  SECTION("InfInArgMin")
+  {
+    const int n     = 10;
+    const float inf = ::cuda::std::numeric_limits<float>::infinity();
+
+    thrust::device_vector<in_t> in(n, inf);
+    const in_t *d_in = thrust::raw_pointer_cast(in.data());
+
+    device_arg_min(d_in, d_out, n);
+
+    const out_t result = out[0];
+    REQUIRE(result.key == 0);
+    REQUIRE(result.value == inf);
+  }
+
+  /**
+   * ArgMax should return lowest value for empty input. This interferes with
+   * input data containing infinity values. This test checks that ArgMax
+   * works correctly with infinity values.
+   */
+  SECTION("InfInArgMax")
+  {
+    thrust::device_vector<in_t> in(n, -inf);
+    const in_t *d_in = thrust::raw_pointer_cast(in.data());
+
+    device_arg_max(d_in, d_out, n);
+
+    const out_t result = out[0];
+    REQUIRE(result.key == 0);
+    REQUIRE(result.value == -inf);
+  }
+}
+
+// TEST_TYPES == 5
+#elif TEST_TYPES == 5
+CUB_TEST("Device reduce works with fancy input iterators", "[reduce][device]", iterator_type_list)
+{
+  using params   = params_t<TestType>;
+  using item_t   = typename params::item_t;
+  using output_t = typename params::output_t;
+  using offset_t = int32_t;
+
+  constexpr int max_items    = 5000000;
+  constexpr int min_items    = 1;
+  constexpr int num_segments = 1;
+
+  // Generate the input sizes to test for
+  const int num_items = GENERATE_COPY(take(3, random(min_items, max_items)),
+                                      values({
+                                        min_items,
+                                        max_items,
+                                      }));
+
+  // Prepare input data
+  item_t default_constant{};
+  init_default_constant(default_constant);
+  auto in_it = thrust::make_constant_iterator(default_constant);
+
+  SECTION("reduce")
```

Comment:
Is there any point in a single section per test that I'm missing?

Reply:
I agree, I don't think having a single section is justified here.